engine/runtime/src/test/scala/org/enso/compiler/test/pass/analyse/DataflowAnalysisTest.scala

```diff
@@ -1100,25 +1092,15 @@ class DataflowAnalysisTest extends CompilerTest {
       val lam = ir.asInstanceOf[IR.Function.Lambda]
       val argX =
         lam.arguments.head.asInstanceOf[IR.DefinitionArgument.Specified]
-      val xUse = lam.body.asInstanceOf[IR.Name.Literal]
 
       // The IDs
-      val lamId  = mkStaticDep(lam.getId)
       val argXId = mkStaticDep(argX.getId)
-      val xUseId = mkStaticDep(xUse.getId)
 
       // The info
-      val dependents   = depInfo.dependents
       val dependencies = depInfo.dependencies
 
-      // The test for dependents
-      dependents.getDirect(argXId) shouldEqual Some(Set(xUseId))
```

Comment:
Out of curiosity: Why the changes in this test have to be made? Is it because of changes in `DemandAnalysis`?

Reply:
Yes, it's a horrible terrible spec and honestly I had to stop myself nuking the whole file. So the problem here is, these stupid pass specs use previous passes to build the initial test state. So, demand analysis runs before this, so the input IR for this spec changes and now it blows up. We should discuss this, but IMO in the near future we need to either nuke all of these specs, or rewrite them to be self-sufficient.

Reply:
Isn't the `IR` serializable? If so, we can serialize the input into a file and read it.